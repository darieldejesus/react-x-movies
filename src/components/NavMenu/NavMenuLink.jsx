import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'styled-components';
import { NavLink } from 'react-router-dom';

const Link = Styled(NavLink)`
  color: #63676f;
  font-size: 15px;
  font-weight: 500;
  text-decoration: none;
  text-transform: uppercase;
`;

Link.displayName = 'Link';

const NavItem = Styled.li`
  background: #eaeaeb;
  box-shadow: inset -1px 0px 0px #cecece;
  cursor: pointer;
  display: flex;
  flex-direction: column;
  flex: 1;
  justify-content: center;
  list-style-type: none;
  padding: 10px 20px;
  text-align: center;

  &:last-child {
    box-shadow: none;	
  }`;

NavItem.displayName = 'NavItem';

const NavMenuLink = ({ href, label }) => (
  <NavItem>
    <Link to={href} activeClassName="active">
      {label}
    </Link>
  </NavItem>
);

NavMenuLink.propTypes = {
  href: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
};

export default NavMenuLink;