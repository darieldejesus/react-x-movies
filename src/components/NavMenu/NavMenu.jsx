import React from 'react';
import Styled from 'styled-components';

import NavMenuLinks from '../../config/navigationMenu';
import NavMenuLink from './NavMenuLink';

const NavContainer = Styled.ul`
  background: #ed0105;
  display: flex;
  height: 80px;
  margin: 0;
  padding: 10px 0 0 0;
  position: relative; 
  top: -10px;
  width: 100%;
`;

NavContainer.displayName = 'NavContainer';

const NavMenu = () => (
  <NavContainer>
    {NavMenuLinks.map(section => (
      <NavMenuLink key={section.href} href={section.href} label={section.label} />
    ))}
  </NavContainer>
);

export default NavMenu;
