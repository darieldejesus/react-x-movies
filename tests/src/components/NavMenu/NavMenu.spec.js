import React from 'react';
import 'jest-styled-components';

import NavMenu from '../../../../src/components/NavMenu/NavMenu';

describe('NavMenu', function() {
  beforeEach(() => {
    this.wrapper = shallow(<NavMenu />).dive();
  });

  test('should renders navigation elements', () => {
    expect(toJson(this.wrapper)).toMatchSnapshot();
  });
});
