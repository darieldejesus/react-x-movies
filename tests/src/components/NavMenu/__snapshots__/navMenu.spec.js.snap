// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavMenu should renders navigation elements 1`] = `
.c0 {
  background: #ed0105;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 80px;
  margin: 0;
  padding: 10px 0 0 0;
  position: relative;
  top: -10px;
  width: 100%;
}

<ul
  className="c0"
>
  <NavMenuLink
    href="/"
    key="/"
    label="Home"
  />
  <NavMenuLink
    href="/popular"
    key="/popular"
    label="Most Popular"
  />
  <NavMenuLink
    href="/genres"
    key="/genres"
    label="Genres"
  />
  <NavMenuLink
    href="/reviews"
    key="/reviews"
    label="Reviews"
  />
  <NavMenuLink
    href="/contact"
    key="/contact"
    label="Contact"
  />
</ul>
`;
